### ISSUE TITLE
candle_pii: real_model_detects_email still fails after classifier.bias fix — suspected candle-transformers relative-attention accuracy issue

### ISSUE THREAD

**Issue by bug-ops** [bug-ops/zeph#5457]
## Description

Follow-up to #5443. That issue's originally-filed root cause (`extract_bio_spans` allegedly unable to decode IO-only BIO label schemes) was investigated and found to be **false** — the decode logic already handled it correctly, proven by a pre-existing unit test (`bio_extraction_orphan_i_starts_span`) predating #5443.

During that investigation a real, distinct bug was found and fixed: `candle_transformers::models::debertav2::DebertaV2NERModel::load` (third-party crate `candle-transformers` v0.11.0) builds its classification head with `candle_nn::linear_no_bias`, silently dropping the `classifier.bias` tensor that the real production checkpoint (`iiiorg/piiranha-v1-detect-personal-information`) actually has — confirmed via direct inspection of `model.safetensors`'s header (`classifier.weight` shape `[18,768]` **and** `classifier.bias` shape `[18]` both present). The fix (landing alongside this issue) replaces `DebertaV2NERModel` with a local `PiiNerHead` struct in `crates/zeph-llm/src/classifier/candle_pii.rs` that composes the crate's public `DebertaV2Model` backbone with a bias-including `candle_nn::linear` classifier head.

**However, this fix does not fully resolve #5443's original symptom.** The ignored integration test `real_model_detects_email` still fails after the bias fix, against the real cached model.

## Evidence

Per-token softmax scores for `detect_pii("Contact John Smith at john@example.com for details.")`, captured via temporary `tracing::debug!` instrumentation (removed before the #5443 fix landed):

```
▁Contact  -> O (0.99998)
▁John     -> O (0.98905)   <- should be I-GIVENNAME, confidently wrong
▁Smith    -> O (0.99098)   <- should be I-SURNAME, confidently wrong
▁at       -> O (0.99996)
▁john     -> I-USERNAME (0.4839)  <- right family, wrong subtype, below the 0.75 threshold
@         -> O (0.95263)   <- should be I-EMAIL, confidently wrong
example   -> O (0.99574)   <- should be I-EMAIL, confidently wrong
.         -> O (0.99552)
com       -> O (0.99993)   <- should be I-EMAIL, confidently wrong
```

Ruled out as causes:
- **Tokenization**: verified correct via `tokenizer.id_to_token` — SentencePiece pieces match expected segmentation.
- **`id2label` mapping**: verified correct against the real `config.json` — indices 0-16 are the 17 `I-<TYPE>` labels, 17 is `O`; matches observed `label_idx=17` for "O" predictions and `label_idx=15` for the one `I-USERNAME` hit.
- **Classifier bias**: confirmed loaded (no `VarBuilder` key-not-found error at load time; if `classifier.bias` were missing, `candle_nn::linear`'s `vb.get_with_hints(out_dim, "bias", ...)` would error).

The predictions are **confidently wrong**, not near-uniform noise (a genuinely broken forward pass — e.g. wrong tensor shape or NaNs propagating — would produce ~1/18 ≈ 0.056 confidence per class, not 0.98-0.999). This pattern is consistent with the model's disentangled/relative self-attention not delivering usable contextual signal, causing the classifier to fall back toward the majority class (`O` is the overwhelming majority label in NER training data) rather than genuinely reading context. The one exception (`▁john` → weak `I-USERNAME`) is explainable by a unigram-level prior (the string "john" alone correlates with username-like PII in training data) rather than genuine attention-based reasoning.

## Root-cause hypothesis (unverified)

This checkpoint (`microsoft/mdeberta-v3-base` backbone per `config.json`'s `_name_or_path`) relies **entirely** on relative position information:

```
position_biased_input: false
relative_attention: true
position_buckets: 256
share_att_key: true
pos_att_type: ["p2c", "c2p"]
max_relative_positions: -1
type_vocab_size: 0
```

Since `position_biased_input=false`, `DebertaV2Embeddings::load` never builds absolute position embeddings at all (`candle-transformers-0.11.0/src/models/debertav2.rs`, ~lines 141-149). **All** contextual/positional signal has to come through the disentangled attention's `p2c`/`c2p` relative-position bias terms (`build_relative_position` / `make_log_bucket_position`, `candle-transformers-0.11.0/src/models/debertav2.rs` ~lines 1364-1420, used inside `DisentangledSelfAttention`). If that bucketed relative-position attention has a subtle bug for this exact config shape (`position_buckets=256`, `share_att_key=true`), the model would behave exactly as observed: it still "runs" (no panics, no NaNs, no shape errors) and produces confident but context-blind predictions, defaulting to the per-token/majority-class prior instead of true NER behavior.

**This hypothesis is unverified.** The investigating environment had no `torch`/`transformers` Python install (`ModuleNotFoundError` for both), so there was no way to diff intermediate hidden states or attention weights against the real HuggingFace reference implementation.

## Reproduction Steps

1. `cargo nextest run -p zeph-llm --features classifiers -E 'test(real_model_detects_email)' --run-ignored ignored-only`
2. Model must be cached (`~/.cache/huggingface/hub/models--iiiorg--piiranha-v1-detect-personal-information`) or downloadable.
3. Observe: test fails, `has_pii: false`, `spans: []`, even with the #5443 `PiiNerHead` bias fix applied.

## Expected Behavior

`real_model_detects_email` passes — the model correctly identifies `John`/`Smith` as name PII and the email address as `EMAIL` PII with confidence above the 0.75 threshold.

## Actual Behavior

Model produces confident `O` predictions for all textbook PII tokens except a weak, sub-threshold signal on one token.

## Suggested Next Step

Requires an environment with `torch` + `transformers` available to numerically diff `DebertaV2Encoder`'s per-layer hidden states (or at minimum final logits) between `candle-transformers`' Rust implementation and the Python HuggingFace reference, for the same tokenized input, to pinpoint whether the bug is in the relative-position bucketing, the disentangled attention combination, or elsewhere in the encoder stack. Alternatively: file upstream against `candle-transformers` if a `candle-transformers` maintainer can confirm/deny known issues with `mdeberta-v3-base`-family checkpoints.

## Environment

- Version: 0.21.4, HEAD 37d73210 (post #5446)
- Features: classifiers (candle, zeph-llm/classifiers)
- Model: `iiiorg/piiranha-v1-detect-personal-information` (mDeBERTa-v3-base backbone)
- `candle-transformers` 0.11.0, `candle-nn` 0.11.0, `candle-core` 0.11.0

## Related

- #5443 (original issue; landing as a partial fix — the classifier-bias bug described there was real and is fixed, but this issue tracks the remaining, deeper symptom)

**Comment by bug-ops** [comment]
## Root cause found: not a code bug — test defect exercising a genuine (harmless) checkpoint limitation

Resolved in #5529.

Two independent code audits (line-by-line review of `candle-transformers` v0.11.0's DeBERTa-v2 disentangled-attention implementation against the published reference algorithm, and a `model.safetensors` header/key-mapping audit) found no defect in the candle port or this crate's loading code.

**Conclusive evidence**: a numeric diff against a real `torch`+`transformers` reference implementation (same checkpoint, same input sentence) showed the candle port's per-token logits match the reference implementation exactly, to 4 decimal places, on every token. The `candle-transformers` DeBERTa-v2 port is numerically correct.

The actual finding: `iiiorg/piiranha-v1-detect-personal-information` is genuinely weak at free-text given-name/surname/email recognition in casual sentences — even in the original HuggingFace/PyTorch implementation — while reliably flagging structured PII (SSNs, street addresses, phone numbers) with >0.99 confidence. The original `real_model_detects_email` test exercised the model's weak spot, not a code defect.

**Fix applied**:
- Replaced `real_model_detects_email` with `real_model_detects_ssn`, using an input verified against the real model to produce confident, stable detection.
- Added a characterization test (`free_text_names_yield_no_confident_span`) that pins the checkpoint's current weak-spot behavior on the original sentence, so a future change (checkpoint bump, candle-transformers upgrade, threshold change) that shifts this behavior in either direction is caught by CI instead of silently drifting.
- The full investigation trail (both audits plus the numeric diff) is preserved in the `PiiNerHead` doc comment in `crates/zeph-llm/src/classifier/candle_pii.rs` for future reference.

**Known limitation, tracked separately**: the confirmed weakness on free-text personal names has no compensating regex fallback (unlike EMAIL/PHONE/SSN/CREDIT_CARD, which are also covered by `zeph-sanitizer`'s regex `PiiFilter`) — filed as a follow-up issue, not a blocker for this fix.
